### ISSUE TITLE
Reimplement ManagedIdentityCredential with composition

### ISSUE THREAD

**Issue by chlowell** [Azure/azure-sdk-for-python#9302]
`ManagedIdentityCredential` exists to hide `MsiCredential` and `ImdsCredential` behind a facade that has the right surface for documentation and code completion. It accomplishes this by overriding its constructor to return an instance of one of those classes, and defining empty methods with docstrings.

It could instead hold another credential and wrap its methods, as does `EnvironmentCredential`.

**Comment by HankiGreed** [comment]
I want to take this issue for my first contribution.

**Comment by chlowell** [comment]
The gist is that `ManagedIdentityCredential` is currently a factory that can't itself be instantiated.  It overrides `__new__` to return one of the implementation credentials. So, you call `ManagedIdentityCredential()` and get something else:
```py
>>> ManagedIdentityCredential()
<azure.identity._credentials.managed_identity.ImdsCredential object at 0x011EAF88>
```
`ManagedIdentityCredential` could instead have an ordinary `__init__` and hold an instance of the appropriate implementation credential. It could implement `get_token` by wrapping that credential's `get_token` method, similarly to `EnvironmentCredential`. I suggest starting with a look at `EnvironmentCredential`.

**Comment by chlowell** [comment]
Addressed by #9302, thanks @HankiGreed!

**Comment by HankiGreed** [comment]
Thanks for giving me a chance to do this. This is my first contribution on GitHub and it felt good doing it. Thanks @chlowell for bearing with me throughout. !
